tensorboard/plugins/hparams/api.py

```diff
@@ -14,7 +14,85 @@
 # ==============================================================================
 """Experimental public APIs for the HParams plugin.
 
-These are porcelain on top of `api_pb2` (`api.proto`) and `summary.py`.
+This module supports a spectrum of use cases, depending on how much
+structure you want. In the simplest case, you can simply collect your
+hparams into a dict, and use a Keras callback to record them:
+
+>>> from tensorboard.plugins.hparams import api as hp
+>>> hparams = {
+...     "optimizer": "adam",
+...     "fc_dropout": 0.2,
+...     "neurons": 128,
+...     # ...
+... }
+>>>
+>>> model = model_fn(hparams)
+>>> callbacks = [
+>>>     tf.keras.callbacks.TensorBoard(logdir),
+>>>     hp.KerasCallback(logdir, hparams),
+>>> ]
+>>> model.fit(..., callbacks=callbacks)
+
+To control how hyperparameters and metrics appear in the TensorBoard
+frontend, you can define `HParam` and `Metric` objects and collect them
+in an `Experiment`:
+
+>>> HP_OPTIMIZER = hp.HParam("optimizer")
+>>> HP_FC_DROPOUT = hp.HParam(
+...     "fc_dropout",
+...     display_name="f.c. dropout",
+...     description="Dropout rate for fully connected subnet.",
+... )
+>>> HP_NEURONS = hp.HParam("neurons", description="Neurons per dense layer")
+>>>
+>>> experiment = hp.Experiment(
+...     hparams=[
+...         HP_OPTIMIZER,
+...         HP_FC_DROPOUT,
+...         HP_NEURONS,
+...     ],
+...     metrics=[
+...         hp.Metric("xent", group="validation", display_name="cross-entropy"),
+...         hp.Metric("f1", group="validation", display_name="F&#x2081; score"),
+...         hp.Metric("loss", group="train", display_name="training loss"),
+...     ],
+... )
+>>> with tf.summary.create_file_writer(base_logdir).as_default():
+...   hp.experiment(experiment)  # write experiment summary
```

Comment:
Does this only support TF2? If so, is it true for the Keras callback and should we clarify that above?

Reply:
The `hp.experiment` endpoint supports TF 1.x and 2.x both; this is
tested in #2139. The Keras callback indeed only supports TF 2.x. I’ll
clarify that above. Thanks!

Reply:
More precisely, the Keras callback only supports TF eager. You can use
it in V1 if you explicitly enable eager mode.